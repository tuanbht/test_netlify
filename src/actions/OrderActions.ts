import { CANCEL_ORDER_PATH, MARK_AS_PAID_PATH, ORDER_DETAILS_PATH } from '../constants/ApiPaths';
import { replace } from 'lodash';
import { Action } from 'redux';
import { CANCEL_ORDER, GET_ORDER_DETAILS, MARK_AS_PAID_ORDER } from '../constants/ReduxActions';
import { get, put } from './AxiosRequestActions';

const buildUrl = (url: string, orderId: number) => {
  return replace(url, ':id', String(orderId));
};

const getOrderDetailsAction = (orderId: number): Action => {
  return get(GET_ORDER_DETAILS, buildUrl(ORDER_DETAILS_PATH, orderId));
};

const cancelOrderAction = (orderId: number): Action => {
  return put(CANCEL_ORDER, buildUrl(CANCEL_ORDER_PATH, orderId));
};

const markAsPaidOrderAction = (orderId: number): Action => {
  return put(MARK_AS_PAID_ORDER, buildUrl(MARK_AS_PAID_PATH, orderId));
};

export default {
  getOrderDetailsAction,
  cancelOrderAction,
  markAsPaidOrderAction,
};
