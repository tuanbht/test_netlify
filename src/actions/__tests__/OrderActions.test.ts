import OrderActions from '../OrderActions';

describe('OrderActions', () => {
  describe('getOrderDetailsAction', () => {
    it('returns get order details action', () => {
      expect(OrderActions.getOrderDetailsAction(1)).toMatchSnapshot();
    });
  });

  describe('cancelOrderAction', () => {
    it('returns cancel order action', () => {
      expect(OrderActions.cancelOrderAction(1)).toMatchSnapshot();
    });
  });

  describe('markAsPaidOrderAction', () => {
    it('returns mark as paid order action', () => {
      expect(OrderActions.markAsPaidOrderAction(1)).toMatchSnapshot();
    });
  });
});
