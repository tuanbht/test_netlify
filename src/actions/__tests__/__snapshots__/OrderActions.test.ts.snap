// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderActions cancelOrderAction returns cancel order action 1`] = `
Object {
  "payload": Object {
    "request": Object {
      "data": Object {},
      "method": "patch",
      "url": "/orders/1/payment/cancel",
    },
  },
  "type": "CANCEL_ORDER",
}
`;

exports[`OrderActions getOrderDetailsAction returns get order details action 1`] = `
Object {
  "payload": Object {
    "request": Object {
      "method": "get",
      "params": Object {},
      "url": "/orders/1/payment",
    },
  },
  "type": "GET_ORDER_DETAILS",
}
`;

exports[`OrderActions markAsPaidOrderAction returns mark as paid order action 1`] = `
Object {
  "payload": Object {
    "request": Object {
      "data": Object {},
      "method": "patch",
      "url": "/orders/1/payment/mark_as_paid",
    },
  },
  "type": "MARK_AS_PAID_ORDER",
}
`;
